### ISSUE TITLE
Feature Reqeust: Full HD Resolution

### ISSUE THREAD

**Issue by regnets** [CatxFish/obs-virtual-cam#84]
Hi CatxFish,

TL;DR: VirtualCam is great, i would like to have Full HD and would like to pay for that. ;)

i am currently using your great Plugin for Video Conferencing Services like Teams/Webex and Gotomeeting. For the sake of corporate identity i added our company Logo to my OBS screen. However the logo and text is really blocky due to the convert from 1080p to 720p. I really would appreciate a version of virtual cam with full hd support. 

I found a few threads scattered arround the internet which all got replied to that request with "Virtual Cam just support 720p, get used to it". However i didn't find an issue or a feature request for Full HD resolution.

Could you please explain why this is not possible? That would help me and many others to understand and to accept it.

If this is something you could fix in software i would like to add a bounty to this issue.

Thanks a lot.

Cheers

Andy

**Comment by musm** [comment]
Yeah totally agree, would love this as well

**Comment by ilmatematico** [comment]
VirtualCam outputs at 1080p@30fps. Maybe your problem is on the other side of the chain. Usually, Webcam Softwares (Skype included) are limited to HD (720p). Doublecheck if your software allow for Full HD sources.

**Comment by arpadarpa** [comment]
> 
> 
> VirtualCam already fully supports outputs at 1080p@30fps. Maybe your problem is on the other side of the chain. Usually, Webcam Softwares are limited to HD (720p). Doublecheck if your software allows for Full HD sources.
> 
> Webex, Gotomeeting and Teams are all bounded to 720p as far as I now

Microsoft Teams supports 1080p. When a direct source is selected (like a Elgato Camlink 4K in my case or Snap Cam) the resolution is 1080p. Using Virtual Cam the resolution is 720p - or lower? - and the down-scaling results in a very pixelated picture. See sceenshot from Teams Client: [https://1drv.ms/u/s!AmEXkbIw6UPUgdNWWvKj3KLBM2psjg?e=dLBXFH](url)

**Comment by regnets** [comment]
As Teams doesn't have any settings in the devices tab i think you have to nail your settings down before starting teams? Or we could have an option to restrict Virtual Cam to 1080p?

**Comment by Algram** [comment]
I can confirm that it looks the same in MS Teams when using it with the Linux compatible plugin: https://github.com/CatxFish/obs-v4l2sink

**Comment by Herbertsz** [comment]
I have the same problem with MS Teams and the virtual camera plugin. Tried several settings, on OBS and Teams side, but of no help.

**Comment by SplatManDK** [comment]
I trust you are all aware that Microsoft has capped Teams to 720p due to the increased usage/load during the COVID-19 crisis? 

https://mspoweruser.com/microsoft-service-restrictions-office-365-demand/

Restrictions to Teams are imposed as a general policy. It's 720p until the COVID-19 crisis is over, or until Microsoft has sufficient capacity to meet demands.

**Comment by tsandner** [comment]
I did some tests comparing comparing OBS Virtual Cam in Teams with vMix and vMix is much better. I.e. text is shown much clearer etc. in exactly the same setup.

**Comment by arpadarpa** [comment]
> 
> 
> I trust you are all aware that Microsoft has capped Teams to 720p due to the increased usage/load during the COVID-19 crisis?
> 
> https://mspoweruser.com/microsoft-service-restrictions-office-365-demand/
> 
> Restrictions to Teams are imposed as a general policy. It's 720p until the COVID-19 crisis is over, or until Microsoft has sufficient capacity to meet demands.

No. Only the recordings are down-scaled to 720p. It applies to Microsoft Steam only.

**Comment by goveebee** [comment]
I'm having the same issue both in Zoom and in Slacks video calls. Any ideas?

**Comment by backbone2k** [comment]
I am also having the issue with the Microsoft Teams desktop client. No text below 16pt is readable. Using MS Teams web client with Edge Chromium shows a FullHD video without issues - switching back to Teams desktop client shows low res picture. https://webcamtests.com/ also works as expected in 1920x1080.  Other video sources including Snap Cam are working.

**Comment by regnets** [comment]
Maybe a build with a fixed resolution to 1920x1080 would be a workarround for the most users?

**Comment by penCsharpener** [comment]
I also have the same problem in MS Teams. When using the NDI virtual cam driver text is crisp and very readable. With OBS VirtualCam it's not. Doesn't even look like 720p.
